lib/app.rb

```diff
@@ -15,58 +15,61 @@ def initialize(stdout, stdin)
     @prompter = Prompter.new(stdout)
   end
 
+  def projects
+    @projects ||= []
+  end
+
+  def current_project
+    @current_project ||= {}
+  end
+
   def run
     menu_printer.project_menu
 
     command = user_input.get_project_input
 
     while command != "q"
-      if !@current_project
+      if current_project.empty?
```

Comment:
This `current_project.empty?` should also be a method instead of accessing the attr accessor directly. If you are waiting to do more work on current project before this kind of extraction, then you should wait until you are ready to do it all.

Reply:
Okay! Moved. Will reach out to you about extracting current_project.